bitflow/processingstep.py

```diff
@@ -146,70 +153,182 @@ def initialize_step(name, script_args):
 
 
 def string_lst_to_lst(str_lst):
-    values = str_lst.split(STRING_LIST_SEPERATOR)
+    values = str_lst.split(STRING_LIST_SEPARATOR)
     return [x.strip() for x in values]
 
 
-SUBCLASSES_TO_IGNORE = ["AsyncProcessingStep",
-                        "Fork"]
+SUBCLASSES_TO_IGNORE = ["AsyncProcessingStep", "Fork", "BatchProcessingStep", "PipelineTermination",
+                        "BatchPipelineTermination"]
 
 
-class ProcessingStep:
+class ProcessingStep(metaclass=helper.CtrlMethodDecorator):
     """ Abstract ProcessingStep Class"""
-    subclasses = []
     __description__ = "No description written yet."
 
     def __init__(self):
+        self.__name__ = "ProcessingStep"
+        self.lock = multiprocessing.Lock()
```

Comment:
this does not inherit from multiprocessing, why is this necessary here? Sinksteps defining their own threading.lock